### PR TITLE
authorization without read + accept-language

### PR DESCRIPTION
**Pull request by maartenRamadan**
#52
#47

### CODE REVIEW

File: src/config/blocks/RouteBlock.cpp [hunk 1]
```diff
@@ -80,10 +80,11 @@ const AConfigBlock::validatorsMapType	RouteBlock::_validators =
 		  .add(new ArgumentLength(1))
 		  .add(new Unique())
 		  .build())
-		.addKey("auth_basic_user_file", ConfigValidatorListBuilder()
-		  .add(new ArgumentLength(1))
+		.addKey("authorized", ConfigValidatorListBuilder()
+		  .add(new Unique())
+		  .build())
```
Comment (mrjvs) [inline src/config/blocks/RouteBlock.cpp]: add the `Required("auth_basic", true)` validator (which makes it so this key requires the other key to function. (add it the other way around too)

File: src/server/http/HTTPParser.cpp [hunk 2]
```diff
@@ -197,6 +198,46 @@ HTTPParser::ParseReturn		HTTPParser::parseHeaders(HTTPParseData &data, const std
...
 
+	// TODO only accept charset UTF-8
+	if (data._type == HTTPParseData::REQUEST) {
+		std::map<std::string,std::string>::iterator acceptCharsets = data.headers.find("ACCEPT-CHARSETS");
+		if (acceptCharsets != data.headers.end() && (acceptCharsets->second != "utf-8" || acceptCharsets->second != "*")) {
+			globalLogger.logItem(logger::ERROR, "Charset not accepted");
+			data.parseStatusCode = 406; // 406 not accepted
+			return ERROR;
+		}
+	}
```
Comment (mrjvs) [inline src/server/http/HTTPParser.cpp]: dont handle

**Review by pmerku — APPROVED** [review]

File: src/server/http/HTTPResponder.cpp [hunk 3]
```diff
@@ -194,7 +194,7 @@ void HTTPResponder::prepareFile(HTTPClient &client, config::ServerBlock &server,
 		return;
 	}
 
-	client.data.response.builder.setHeader("Content-Type", MimeTypes::getMimeType(file.getExt()));
+	client.data.response.builder.setHeader("Content-Type", MimeTypes::getMimeType(file.getExt())); // TODOHEADERS + charset=utf-8 ?
```
Comment (mrjvs) [inline src/server/http/HTTPResponder.cpp]: remove todo, not needed

File: src/server/http/HTTPResponder.cpp [hunk 4]
```diff
@@ -245,6 +245,32 @@ void HTTPResponder::serveFile(HTTPClient &client, config::ServerBlock &server, c
...
+				if (requestLanguages->second == *acceptedLanguages) {
+					client.data.request.data.acceptLanguage = requestLanguages->second;
+					break ;
+				}
+			}
+			if (!client.data.request.data.acceptLanguage.empty())
+				break ;
+		}
+	}
+	// error if no match was found
```
Comment (mrjvs) [inline src/server/http/HTTPResponder.cpp]: put in !empty if statement

**Review by mrjvs — CHANGES_REQUESTED** [review]

**Review by mrjvs — APPROVED** [review]

**Pull request merged**